### PR TITLE
Intellij Cleanup

### PR DESCRIPTION
**Pull request by yschimke**

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/HttpUrl.kt [hunk 1]
```diff
@@ -1144,15 +1162,17 @@ class HttpUrl internal constructor(builder: Builder) {
...
-        if (input.regionMatches(pos, "https:", 0, 6, ignoreCase = true)) {
-          this.scheme = "https"
-          pos += "https:".length
-        } else if (input.regionMatches(pos, "http:", 0, 5, ignoreCase = true)) {
-          this.scheme = "http"
-          pos += "http:".length
-        } else {
-          throw IllegalArgumentException("Expected URL scheme 'http' or 'https' but was '"
-              + input.substring(0, schemeDelimiterOffset) + "'")
+        when {
```
Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/HttpUrl.kt]: Not sure about this one. How about
```kotlin
scheme = when {
  ...
}
pos += scheme.length
```

File: okhttp/src/main/java/okhttp3/HttpUrl.kt [hunk 2]
```diff
@@ -1368,17 +1388,17 @@ class HttpUrl internal constructor(builder: Builder) {
...
-              || c == '-'
-              || c == '.') {
-            continue // Scheme character. Keep going.
+          return if (c in 'a'..'z'
+                  || c in 'A'..'Z'
+                  || c in '0'..'9'
+                  || c == '+'
+                  || c == '-'
+                  || c == '.') {
+              continue // Scheme character. Keep going.
```
Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/HttpUrl.kt]: This one I think I don't agree with. If we're hoisting a `return` from an `if` I would prefer it only return or throw and not mix other control flows like `continue`.

File: okhttp/src/main/java/okhttp3/Cache.kt [hunk 3]
```diff
@@ -506,7 +506,7 @@ class Cache internal constructor(
      */
     @Throws(IOException::class)
     internal constructor(rawSource: Source) {
-      try {
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/Cache.kt]: Flagged in build

```
> Task :okhttp:compileKotlin FAILED
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (443, 5): Property must be initialized or be abstract
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (444, 5): Property must be initialized or be abstract
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (445, 5): Property must be initialized or be abstract
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (446, 5): Property must be initialized or be abstract
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (447, 5): Property must be initialized or be abstract
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (448, 5): Property must be initialized or be abstract
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (449, 5): Property must be initialized or be abstract
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (450, 5): Property must be initialized or be abstract
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (451, 5): Property must be initialized or be abstract
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (452, 5): Property must be initialized or be abstract
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (511, 9): Captured member values initialization is forbidden due to possible reassignment
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (512, 9): Captured member values initialization is forbidden due to possible reassignment
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (518, 9): Captured member values initialization is forbidden due to possible reassignment
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (521, 9): Captured member values initialization is forbidden due to possible reassignment
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (522, 9): Captured member values initialization is forbidden due to possible reassignment
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (523, 9): Captured member values initialization is forbidden due to possible reassignment
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (533, 9): Captured member values initialization is forbidden due to possible reassignment
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (534, 9): Captured member values initialization is forbidden due to possible reassignment
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (535, 9): Captured member values initialization is forbidden due to possible reassignment
e: /home/circleci/project/okhttp/src/main/java/okhttp3/Cache.kt: (551, 11): Captured member values initialization is forbidden due to possible reassignment
```

File: okhttp/src/main/java/okhttp3/HttpUrl.kt (hunk 1 above)
Comment (yschimke) [reply okhttp/src/main/java/okhttp3/HttpUrl.kt]: scheme is changeable in between, so ended up with an awkward also, or a !! check

Comment (JakeWharton) [reply okhttp/src/main/java/okhttp3/HttpUrl.kt]: Maybe read to local?
```kotlin
val scheme = when {
  // ...
}
this.scheme = scheme
pos += scheme.length
```

File: okhttp/src/main/java/okhttp3/HttpUrl.kt [hunk 4]
```diff
@@ -1143,22 +1161,18 @@ class HttpUrl internal constructor(builder: Builder) {
...
-        }
-      } else if (base != null) {
-        this.scheme = base.scheme
-      } else {
-        throw IllegalArgumentException(
-            "Expected URL scheme 'http' or 'https' but no colon was found")
+      when {
+          schemeDelimiterOffset != -1 -> {
+            this.scheme = when {
+              input.startsWith("https:", ignoreCase = true) -> "https"
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/HttpUrl.kt]: `input.startsWith("https:", ignoreCase = true)` is nice

File: okhttp/src/main/java/okhttp3/HttpUrl.kt [hunk 5]
```diff
@@ -838,24 +838,28 @@ class HttpUrl internal constructor(builder: Builder) {
...
-      this.encodedPassword = canonicalize(password, PASSWORD_ENCODE_SET, false, false, false, true)
+      this.encodedPassword = canonicalize(password, PASSWORD_ENCODE_SET, alreadyEncoded = false,
+              strict = false, plusIsSpace = false, asciiOnly = true)
       return this
     }
 
     fun encodedPassword(encodedPassword: String): Builder {
       this.encodedPassword = canonicalize(
-          encodedPassword, PASSWORD_ENCODE_SET, true, false, false, true)
+          encodedPassword, PASSWORD_ENCODE_SET, alreadyEncoded = true, strict = false,
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/HttpUrl.kt]: these all get even nicer if we provide sane defaults

File: okhttp/src/main/java/okhttp3/HttpUrl.kt [hunk 6]
```diff
@@ -1371,9 +1389,9 @@ class HttpUrl internal constructor(builder: Builder) {
         for (i in pos + 1 until limit) {
           val c = input[i]
 
-          if (c >= 'a' && c <= 'z' ||
-              c >= 'A' && c <= 'Z' ||
-              c >= '0' && c <= '9' ||
+          if (c in 'a'..'z' ||
+                  c in 'A'..'Z' ||
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/HttpUrl.kt]: wrapping looks off

File: okhttp/src/main/java/okhttp3/internal/io/FileSystem.kt [hunk 7]
```diff
@@ -50,24 +50,24 @@ interface FileSystem {
...
-          // Maybe the parent directory doesn't exist? Try creating it first.
-          file.parentFile.mkdirs()
-          return file.sink()
-        }
+          return try {
+              file.sink()
+          } catch (_: FileNotFoundException) {
+              // Maybe the parent directory doesn't exist? Try creating it first.
+              file.parentFile.mkdirs()
+              file.sink()
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/internal/io/FileSystem.kt]: the only case I don’t really like is this one, which doesn’t jump out at me as being used as the return value

**Pull request merged**